src/node_dotenv.cc

```diff
@@ -101,35 +92,127 @@ Local<Object> Dotenv::ToObject(Environment* env) {
   return result;
 }
 
-void Dotenv::ParseContent(const std::string_view content) {
-  std::string lines = std::string(content);
-  lines = std::regex_replace(lines, std::regex("\r\n?"), "\n");
+std::string_view trim_spaces(std::string_view input) {
+  if (input.empty()) return "";
+  if (input.front() == ' ') {
+    input.remove_prefix(input.find_first_not_of(' '));
+  }
+  if (!input.empty() && input.back() == ' ') {
+    input = input.substr(0, input.find_last_not_of(' ') + 1);
+  }
+  return input;
+}
+
+void Dotenv::ParseContent(const std::string_view input) {
+  std::string_view content = input;
+
+  std::string_view key;
+  std::string_view value;
 
-  std::smatch match;
-  while (std::regex_search(lines, match, LINE)) {
-    const std::string key = match[1].str();
+  content = trim_spaces(content);
+
+  while (!content.empty()) {
+    // Skip empty lines and comments
+    if (content.front() == '\n' || content.front() == '#') {
+      auto newline = content.find('\n');
+      if (newline != std::string_view::npos) {
+        content.remove_prefix(newline + 1);
+        continue;
+      }
+    }
+
+    // If there is no equal character, then ignore everything
+    auto equal = content.find('=');
+    if (equal == std::string_view::npos) {
+      break;
+    }
 
-    // Default undefined or null to an empty string
-    std::string value = match[2].str();
+    key = content.substr(0, equal);
+    content.remove_prefix(equal + 1);
+    key = trim_spaces(key);
 
-    // Remove leading whitespaces
-    value.erase(0, value.find_first_not_of(" \t"));
+    if (key.empty()) {
+      break;
+    }
 
-    // Remove trailing whitespaces
-    if (!value.empty()) {
-      value.erase(value.find_last_not_of(" \t") + 1);
+    // Remove export prefix from key
+    auto have_export = key.compare(0, 7, "export ") == 0;
+    if (have_export) {
+      key = key.substr(7);
     }
 
-    if (!value.empty() && value.front() == '"') {
-      value = std::regex_replace(value, std::regex("\\\\n"), "\n");
-      value = std::regex_replace(value, std::regex("\\\\r"), "\r");
+    // SAFETY: Content is guaranteed to have at least one character
+    if (content.empty()) {
+      break;
     }
 
-    // Remove surrounding quotes
-    value = trim_quotes(value);
+    // Expand new line if \n it's inside double quotes
+    // Example: EXPAND_NEWLINES = 'expand\nnew\nlines'
+    if (content.front() == '"') {
+      auto closing_quote = content.find(content.front(), 1);
+      value = content.substr(1, closing_quote - 1);
```

Comment:
If closing_quote does not exist, this could fail. We need to check if closing_quote is not npos.